src/bin/pg_autoctl/cli_common.c

```diff
@@ -309,6 +317,81 @@ cli_create_node_getopts(int argc, char **argv,
 				break;
 			}
 
+			case 's':
+			{
+				/* { "ssl-self-signed", no_argument, NULL, 's' }, */
+				if (sslOrNoSSLCount > 0)
+				{
+					errors++;
+					log_error("Using both --no-ssl and --ssl-self-signed "
+							  "is not supported");
+				}
+				++sslOrNoSSLCount;
+
+				LocalOptionConfig.pgSetup.ssl.active = 1;
+				LocalOptionConfig.pgSetup.ssl.createSelfSignedCert = true;
+				log_trace("--ssl-self-signed");
+				break;
+			}
+
+			case 'N':
+			{
+				/* { "no-ssl", no_argument, NULL, 'N' }, */
+				if (sslOrNoSSLCount > 0)
+				{
+					errors++;
+					log_error("Using both --no-ssl and --ssl-self-signed "
+							  "is not supported");
+				}
+				++sslOrNoSSLCount;
+
+				LocalOptionConfig.pgSetup.ssl.active = 0;
+				LocalOptionConfig.pgSetup.ssl.createSelfSignedCert = false;
+				log_trace("--no-ssl");
+				break;
+			}
+
+			/*
+			 * { "ssl-ca-file", required_argument, &ssl_flag, SSL_CA_FILE_FLAG }
+			 * { "ssl-crl-file", required_argument, &ssl_flag, SSL_CA_FILE_FLAG }
+			 * { "server-crt", required_argument, &ssl_flag, SSL_SERVER_CRT_FLAG }
+			 * { "server-key", required_argument, &ssl_flag, SSL_SERVER_KEY_FLAG }
+			 * { "ssl-mode", required_argument, &ssl_flag, SSL_MODE_FLAG },
+			 */
+			case 0:
+			{
+				if (ssl_flag != SSL_MODE_FLAG)
+				{
+					/* we have 4 options allowed to all reach this code */
+					if (!sslUserCertificates)
+					{
+						/*
+						 * but we can't use any of those options together with
+						 * the --no-ssl or the --ssl-self-signed options
+						 */
+						if (sslOrNoSSLCount > 0)
+						{
+							errors++;
+							log_error(
+								"Using either --no-ssl or --ssl-self-signed "
+								"with user-provided SSL certificates "
+								"is not supported");
+						}
+
+						sslUserCertificates = true;
+						++sslOrNoSSLCount;
```

Comment:
this expands the meaning of the variable. it becomes basically any ssl setting count

Reply:
Fixed thanks to the new enum for tracking user choice for SSL setup.